modules/iam__enum_roles/main.py

```diff
@@ -124,7 +153,48 @@ def main(args, pacu_main):
 
                     data['roles_assumed'].append(role)
 
-    return data
+
+def main(args, pacu_main):
+    args = parser.parse_args(args)
+    iam: 'mypy_boto3_iam.IAMClient' = pacu_main.get_boto3_client('iam')
+
+    if args.role_name:
+        role_name = args.role_name
+        resp = iam.get_role(RoleName=role_name)
+        orig_trust_doc = json.dumps(resp['Role']['AssumeRolePolicyDocument'])
+    else:
+        role_name = f"PacuIamEnumRoles-{''.join(random.choice(chars) for _ in range(5))}"
+        iam.create_role(
+            RoleName=role_name,
+            Description="Created by Pacu for IAM Role enumeration.",
+            AssumeRolePolicyDocument='''
+            {
+                "Version": "2012-10-17",
+                "Statement": [{
+                    "Effect": "Deny",
+                    "Principal": {"AWS": "*"},
+                    "Action": "sts:AssumeRole"
+                }]
+            }
+            '''.strip(),
+        )
+
+    try:
+        run(args, role_name, pacu_main, iam)
```

Comment:
Code from main was moved to run so we can cleanup the role if it fails for whatever reason.